### PR TITLE
Fix infinite loop in bulk sync

### PR DESCRIPTION
**Pull request by mrjana**
Due to a slice management logic error the bulk sync for loop can go on
indefinitely and eventually leading to an OOM error. Fixed the logic so
that an infinite loop never occurs. Also changed the bulk sync wait
timeout to use a timer rather than use time.After as time.After is known
to consume a lot of memory when called in a tight loop.

Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>

### CODE REVIEW

**Comment by aboch** [comment]
LGTM

**Comment by mavenugo** [comment]
LGTM

File: networkdb/cluster.go [hunk 1]
```diff
@@ -449,8 +453,9 @@ func (nDB *NetworkDB) bulkSyncNode(networks []string, node string, unsolicited b
 	// Wait on a response only if it is unsolicited.
 	if unsolicited {
 		startTime := time.Now()
+		t := time.NewTimer(30 * time.Second)
 		select {
-		case <-time.After(30 * time.Second):
```
Comment (mavenugo) [inline networkdb/cluster.go]: It looks like you are using `<-t.C` instead of `After`. But both seems identical from a functionality standpoint. Why is this change needed ?

Comment (mrjana) [inline networkdb/cluster.go]: `time.After` consumes much more memory that `Timer`. If you don't expect a timeout to happen a lot then you use the `Timer` then `time.After`

**Comment by mavenugo** [comment]
@mrjana thanks for the confirmation. I tested it end to end in a setup where I was consistently hitting this issue and this addresses the mem-leak and infinite loop issue.
